src/main/groovy/com/calgaryscientific/gradle/VeracodeDetailedReportCSVTask.groovy

```diff
@@ -26,57 +26,232 @@
+                'CWE ID',
+                'Date',
+                'Description',
+                'Exploit Level',
+                'Function Prototype',
+                'Function Relative Location',
+                'Issue ID',
+                'Line',
+                'Mitigation Status',
+                'Mitigation StatusDesc',
+                'Module',
+                'Remediation Status',
+                'Remediation Effort',
+                'Severity',
+                'Source File',
+                'Source File Path',
+                'Type'
+        ]
+        csvFlawsReport.printRecord(headerRow)
+        csvOpenFlawsReport.printRecord(headerRow)
+        NodeList severityList = XMLIO.getNodeList(xml, 'severity')
+        for (int i = 0; i < severityList.size(); i++) {
+            Node severity = severityList.get(i) as Node
+            NodeList categoryList = XMLIO.getNodeList(severity, 'category')
+            for (int j = 0; j < categoryList.size(); j++) {
+                Node category = categoryList.get(j) as Node
+                /**
+                 * cwe tree:
+                 *
+                 * /cwe @certc @certcpp @certjava @cweid @cwename @owasp @pcirelated @sans
+                 *     /description
+                 *         /text @text
+                 *     /staticflaws
+                 *         /flaw @affects_policy_compliance
+                 *              -@categoryid
+                 *              -@categoryname
+                 *              -@cia_impact
+                 *              -@count
+                 *              -@cweid
+                 *              -@date_first_occurrence
+                 *              -@description
+                 *              -@exploitLevel
+                 *              -@functionprototype
+                 *              -@functionrelativelocation
+                 *              -@grace_period_expires
+                 *              -@issueid
+                 *              -@line
+                 *              -@mitigation_status
+                 *              -@mitigation_status_desc
+                 *              -@module
+                 *              -@note
+                 *              -@pcirelated
+                 *              -@remediation_status
+                 *              -@remediationeffort
+                 *              -@scope
+                 *              -@severity
+                 *              -@sourcefile
+                 *              -@sourcefilepath
+                 *              -@type
+                 *             /mitigations
+                 *                /mitigation @action @date @description @user
+                 *            /annotations
+                 *                /annotation @action @date @description @user
+                 * */
+                NodeList cweList = XMLIO.getNodeList(category, 'cwe')
+                for (int k = 0; k < cweList.size(); k++) {
```

Comment:
I think you can use groovy idiom here, 
```
cweList.each { cweNode ->
...
}
```

Reply:
Groovy idioms don't work on static compilation.